### DISCUSSION TITLE
我与ChatGPT结对编程的体验 · BMPI

### DISCUSSION THREAD

**Discussion by giscus** [bmpi-dev/bmpi.dev#147] [Announcements]
# 我与ChatGPT结对编程的体验 · BMPI

本文分享了笔者与 ChatGPT 结对编程的体验并分享一些 Prompt Engineering 的重要技巧。

https://www.bmpi.dev/dev/chatgpt-development-notes/pair-programming/

**Comment by madawei2699** [comment]
使用多步提示编写单元测试：https://github.com/openai/openai-cookbook/blob/main/examples/Unit_test_writing_using_a_multi-step_prompt.ipynb

**Comment by madawei2699** [comment]
https://mp.weixin.qq.com/s/cAWYScI-J1Zr3rS2et1xqQ

重构：重构在过去是一个极为重要的敏捷实践。这一点看Martin Fowler的经典著作《重构》的流行度就可以知道。重构在大模型时代，在契约明确的情况下，这个实践很可能会有重大变化。换句话说，当重写的成本极低，而且可靠性极高的情况下，小步重构的手法和重要性可能不再重要。

自动化测试：自动化测试是契约的保障，重要性不会降低。

测试驱动开发：经典的测试驱动开发方法会发生变化。更可能的方向：一个是人类用测试表达契约，或者在人工智能的辅助下表达契约，然后人工智能来负责具体的实现。另外，给大家报告一下，作为测试驱动开发的积极实践者，我最近使用测试驱动开发写代码的频率其实在降低——因为测试驱动开发的一个重要作用是降低认知负担，而人工智能已经可以在这方面提供许多帮助。

结对编程：一个好消息，如果你过去就认同结对编程的理念，也熟悉结对编程的方法，现在你一定是如鱼得水，因为你有了一个知识非常渊博的编程助手。人和人的结对编程不再是唯一的模式。更好的模式是和AI结对编程。

持续集成：仍然重要。

整洁代码：最近的体会是，人工智能帮我写出来的代码，比我看到的“专业的人类程序员”写出来的，好太多。整洁代码，作为软件开发的入门级实践，或许真的已经不再那么重要了。相比过去推广起来的困难，训练一个学习能力更强的机器助手，其实容易得多。
